utils/pretty_printers.cc

```diff
@@ -11,16 +11,22 @@
 namespace utils {
 
 std::ostream& operator<<(std::ostream& os, pretty_printed_data_size data) {
-    static constexpr const char *suffixes[] = {" bytes", "kB", "MB", "GB", "TB", "PB"};
-
-    unsigned exp = 0;
-    while ((data._size >= 1000) && (exp < sizeof(suffixes))) {
-        exp++;
-        data._size /= 1000;
+    static constexpr char* suffixes[] = {" bytes", "kB", "MB", "GB", "TB", "PB"};
+
+    const char* suffix = nullptr;
+    uint64_t size = data._size;
+    uint64_t next_size = size;
+    for (auto s : suffixes) {
+        suffix = s;
+        size = next_size;
+        next_size = size / 1000;
+        if (next_size == 0) {
+            break;
+        }
     }
 
-    os << data._size << suffixes[exp];
-    return os;
+    assert(suffix);
```

Comment:
I think this assert is pointless - it can never ever fail, because the loop above always assigns suffix. This was one of the features of the rewritten loop which I didn't like, by the way.
So the assert just checks something that was done a few lines up, I think it's not useful (and goes against my wish never to use assert :-) ).

Reply:
okay. i am removing it. not because i agree with you on "never to use assert". but the thing being asserted is obvious.

> it can never ever fail

well, that's what `assert()` is for.